compiler/ir/ir.objcinterop/src/org/jetbrains/kotlin/ir/objcinterop/ObjCInterop.kt

```diff
@@ -266,9 +266,14 @@ fun IrClass.getExternalObjCClassBinaryName(): String =
         this.getExplicitExternalObjCClassBinaryName()
                 ?: this.name.asString()
 
-fun IrClass.getExternalObjCMetaClassBinaryName(): String =
-        this.getExplicitExternalObjCClassBinaryName()
-                ?: this.name.asString().removeSuffix("Meta")
+fun IrClass.getExternalObjCMetaClassBinaryName(): String {
+    this.getExplicitExternalObjCClassBinaryName()?.let { return it }
+    // External ObjC metaclass is named as `Foo.Componion`, Foo is subclass of NSObject or itself.
+    if (this.isCompanion) {
+        (this.parent as? IrClass)?.let { return it.name.asString() }
```

Comment:
What if the class has an `@ExternalObjCClass(binaryName = )` annotation? Shouldn't we return that value then, as in other cases?

Please add a test for this.

Reply:
If the class has an`@ExternalObjCClass(binaryName = )` annotation,  it returns the binary name immediately.
The purpose of the `isCompanion` code block is also to obtain the class name.

As for test cases, it seems that this usage is not supported yet; I searched the entire `kotlin` repository and found no usage. Are there any examples?

`java.lang.IllegalArgumentException: External objc Classes can't be used this way`

```kotlin
@OptIn(BetaInteropApi::class)
@ExternalObjCClass(binaryName = "Foo")
open class A: NSObject() {
    companion object
}

@OptIn(BetaInteropApi::class)
@ExternalObjCClass(binaryName = "Bar")
open class B: NSObject() {
    companion object
}

B as A.Companion
```

Reply:
> As for test cases, it seems that this usage is not supported yet; I searched the entire kotlin repository and found no usage. Are there any examples?

`@ExternalObjCClass(binaryName = )` is generated by `cinterop` for classes with `objc_runtime_name` attribute. You are not supposed to add this annotation manually to Kotlin code.

>  If the class has an`@ExternalObjCClass(binaryName = )` annotation, it returns the binary name immediately,
The purpose of the `isCompanion` code block is also to obtain the class name.

If an Objective-C class has `objc_runtime_name` attribute, then `getExternalObjCMetaClassBinaryName` should return this name.